src/adapters/nginad.js

```diff
@@ -0,0 +1,180 @@
+var CONSTANTS = require('../constants.json');
+var utils = require('../utils.js');
+var bidfactory = require('../bidfactory.js');
+var bidmanager = require('../bidmanager.js');
+var adloader = require('../adloader');
+
+var defaultPlacementForBadBid = '';
+
+/**
+ * Adapter for requesting bids from NginAd
+ */
+var NginAdAdapter = function NginAdAdapter() {
+
+	var rtbServerDomain = 'placeholder.for.nginad.server.com';
+
+	function _callBids(params) {
+		var nginadBids = params.bids || [];
+		// De-dupe by tagid then issue single bid request for all bids
+		_requestBids(_getUniqueTagids(nginadBids));
+	}
+
+	// filter bids to de-dupe them?
+	function _getUniqueTagids(bids) {
+		var key;
+		var map = {};
+		var PubZoneIds = [];
+		bids.forEach(function(bid) {
```

Comment:
Array.forEach is really slow, we prefer using a straight `for` loop instead > http://jsperf.com/fast-array-foreach